### PR TITLE
fix: changed proposal votes ednpoint #LSC-272

### PR DESCRIPTION
**Pull request by swelf19**

### CODE REVIEW

File: proposals/repository.go [hunk 1]
```diff
@@ -72,7 +75,39 @@ func (r *Repository) fetch(ctx context.Context, status string) ([]*models.GetPro
 	return resp.GetPayload().Proposals, nil
 }
 
-func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*models.GetProposalVotesResultVotes, error) {
+func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*query.VotesOKBodyVotesItems0, error) {
+	var paginationKey strfmt.Base64
+	limit := "1000"
```
Comment (oopcode) [inline proposals/repository.go:80]: what do you think about moving the limit to a constant?

File: proposals/repository.go [hunk 2]
```diff
@@ -72,7 +75,39 @@ func (r *Repository) fetch(ctx context.Context, status string) ([]*models.GetPro
...
+				PaginationLimit: &limit,
+				ProposalID:      strconv.Itoa(proposalID),
+				Context:         ctx,
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get votes for proposal id = %d: %w", proposalID, err)
+		}
+
+		err = resp.GetPayload().Validate(nil)
```
Comment (oopcode) [inline proposals/repository.go:95]: is `resp.GetPayload()` parsing something or is it just a getter for a pre-parsed object? if it's parsing something we might want to do it just once

Comment (swelf19) [reply proposals/repository.go]: just a getter to a struct field

File: proposals/repository.go [hunk 3]
```diff
@@ -72,16 +75,16 @@ func (r *Repository) fetch(ctx context.Context, status string) ([]*models.GetPro
 	return resp.GetPayload().Proposals, nil
 }
 
-func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*models.GetProposalVotesResultVotes, error) {
-	page := 1.0
-	votes := make([]*models.GetProposalVotesResultVotes, 0)
+func (r *Repository) GetVotes(ctx context.Context, proposalID int) ([]*query.VotesOKBodyVotesItems0, error) {
```
Comment (sotnikov-s) [inline proposals/repository.go:78]: I've played around fcd fetching votes endpoints and I see that there doesn't seem to be an endpoint for getting complete votes list for finished proposals:
- `https://fcd.terra.dev/cosmos/gov/v1beta1/proposals/{proposal_id}/votes` works only for voting proposals, for the finished and deposited ones it simply returns nothingness;
- `https://fcd.terra.dev/v1/gov/proposals/{proposal_id}/votes` works up to the very beginning, but returns a lesser amount of votes.

I assumed maybe the first endpoint expands the validator's unvoted delegators votes, but it doesn't.

That sucks, and also that deceives the user of the method now. What do you think of making the name more specific? e.g. GetVotingProposalVotes. Or maybe an explanation comment would be better

**Comment by swelf19** [comment]
i had to modify TestFetchVoting test, removed validate VotingProposals in WithFCD test since right now there is no ongoing proposals and test with public FCD fails

**Review by sotnikov-s — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**